classes/submission/reviewAssignment/Repository.php

```diff
@@ -227,6 +227,39 @@ public function deleteMany(Collector $collector)
         }
     }
 
+    /**
+     * Delete all review assignments for a given context ID.
+     *
+     * @param int $contextId
+     * @return void
+     */
+    public function deleteByContextId($contextId)
+    {
+        // using reviewAssignmentCollector to fetch and delete all review assignments for the context
+        $reviewAssignmentCollector = $this->getCollector();
+        $reviewAssignmentCollector->filterByContextIds([$contextId]);
+        $reviewAssignments = $reviewAssignmentCollector->getMany();
```

Comment:
You can use `getIds` instead of `getMany`, then just iterate through the IDs. That'll use significantly less memory and resources than `getMany`, which will instantiate each object.